master/buildbot/reporters/slack.py

```diff
@@ -0,0 +1,128 @@
+# This file is part of Buildbot.  Buildbot is free software: you can
+# redistribute it and/or modify it under the terms of the GNU General Public
+# License as published by the Free Software Foundation, version 2.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more
+# details.
+#
+# You should have received a copy of the GNU General Public License along with
+# this program; if not, write to the Free Software Foundation, Inc., 51
+# Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+#
+# Copyright Buildbot Team Members
+
+from __future__ import absolute_import
+from __future__ import print_function
+from future.utils import string_types
+
+from twisted.internet import defer
+
+from buildbot import config
+from datetime import datetime
+from humanize import naturaltime
+from buildbot.status.builder import SUCCESS
+from buildbot.util import httpclientservice
+from buildbot.reporters.http import HttpStatusPushBase
+
+SLACK_BASE_URL = "https://hooks.slack.com"
+
+
+class SlackStatusPush(HttpStatusPushBase):
+    """
+    Sends messages to a Slack.io channel when each build finishes with a handy
+    link to the build results.
+    """
+
+    name = "SlackStatusPush"
+
+    def checkConfig(self, endpoint, username="buildbot",
+                    icon="https://buildbot.net/img/nut.png",
+                    notify_on_success=True, notify_on_failure=True,
+                    **kwargs):
+        if not isinstance(endpoint, string_types):
+            config.error('endpoint must be a string')
+        if not isinstance(username, string_types):
+            config.error('username must be a string')
+        if not isinstance(icon, string_types):
+            config.error('icon must be a string')
+        if not isinstance(notify_on_success, bool):
+            config.error('notify_on_success must be a bool')
+        if not isinstance(notify_on_failure, bool):
+            config.error('notify_on_failure must be a bool')
+
+    @defer.inlineCallbacks
+    def reconfigService(self, endpoint, username="buildbot",
+                        icon="https://buildbot.net/img/nut.png",
```

Comment:
Please don't point to our community infra.

Reply:
It requires a URL, so what do you suggest I point to?

Reply:
I would put a default pointing to the buildbotURL, and make a check to make sure people don't point it to buildbot.net

Reply:
I can't find nut.png anywhere in buildbotURL. Can you point me to it?

Reply:
there is $buildbotURL/img/favicon.ico

Reply:
but we can add a larger version by creating it in www/base/src/img/